src/state/__tests__/state-machine/buying.spec.ts

```diff
@@ -63,3 +66,191 @@ testRandomSeed("buying card from animal market when deck is empty", (seed) => {
   expect(stateAfter.animalMarket.table).not.toContainEqual(cardToBuy);
   expect(stateAfter.animalMarket.table).toHaveLength(3);
 });
+
+testRandomSeed("playing animal cards with shared habitats should unlock habitats", (seed) => {
```

Comment:
JFYI. We have some tools for guards testing around buying stuff - https://github.com/helcomsecretariat/ecosfera-baltica-digital/blob/406e93003b5bb4737e0e54b0039fd0d3978dd5bf/src/state/__tests__/guards.spec.ts#L74-L98